### ISSUE TITLE
Function ExecuteDbQuery cause double execution of Queries

### ISSUE THREAD

**Issue by gantonjo-tnm** [grocy/grocy#751]
ExecuteDbStatement($sql) should not be called for queries that change the database contents, since the following call to $pdo->query($sql) duplicates the change order.

```
        /**
	 * @return boolean|\PDOStatement
	 */
	public function ExecuteDbQuery(string $sql)
	{
		$pdo = $this->GetDbConnectionRaw();
		if ($this->ExecuteDbStatement($sql) === true)
		{
			return $pdo->query($sql);
		}

		return false;
	}

```

Correct function should be:

```
  /**
   * @return boolean|\PDOStatement
   */
  public function ExecuteDbQuery(string $sql)
  {
    $pdo = $this->GetDbConnectionRaw();
    $stmt = $pdo->query($sql);
    if ($stmt != false ){
      return $stmt;
    }
    return false;
  }

```

**Comment by berrnd** [comment]
Thanks, but that's not really relevant since `ExecuteDbStatement` is **only** used for database migrations.

[LessQL](https://github.com/morris/lessql) is used for "normal" database access everywhere (beside a handful of places where `ExecuteDbQuery` is used because LessQL is not enough there).

**Comment by gantonjo-tnm** [comment]
Well, Database migrations may fail or at least take longer time than neccessary. Anyway, you are the designer, so you decide whether or not to improve this.

**Comment by berrnd** [comment]
Again: `ExecuteDbStatement` is nowhere used to change database contents, database mgirations change the database schema only (and yes, also here maybe a handful migrations where changing data was/is necessary).

However, thanks for your hint, feel free to provide that as a pull request, but please make sure to test all places where those functions are used. I'll be glad to merge it, but I have no desire to do this myself.

grocy has not the best codebase in terms of that some others call it ugly. But my primary focus for grocy was and is always the end result. It's very relaxing to not think about theoretical problems. grocy does not handle gigabytes (even not megabytes) of data. grocy is not the ERP for a big business company, as long as you don't call your fridge a business. Working on any of the open features requests brings more fun for everyone than some milliseconds faster database migrations. Sorry for the off-topic.

**Comment by gantonjo-tnm** [comment]
Hehe, that's totally fine. I got interested in your project through Home Assistant. However, since I run my Home Assistant in Docker CE on my Centos 7 server, I ran into problems when I wanted to add the Grocy add in to Home Assistant. I tried to install the Docker versions of Grocy, but could not get any of the Docker solutions to work either. So, when trying to install Grocy directly into the web root of my Centos 7 server, it failed due to SQLite version was too old. Even upgrading SQLite to latest version did not help. 

So, guess what, I started to migrate your code to use MySQL/MariaDB, running all your migrations manually to even out the differences between SQlite jargon and MySQL jargon. This is where I discovered the issue I reported. However, the route cause of the problem installing Grocy in my web root may have been a missing line in the .htaccess file. Anyway, I now have a running version of Grocy with MySQL on my Centos 7 server. Just have to check that it works as intented now.
